src/cli/telemetry/telemetry.ts

```diff
@@ -170,20 +176,67 @@ async function getInstalledPackages(sys: d.CompilerSystem, config: d.Config): Pr
       ([k]) => k.startsWith('@stencil/') || k.startsWith('@ionic/') || k.startsWith('@capacitor/')
     );
 
-    packages = packageLockJson
-      ? ionicPackages.map(
-          ([k, v]) =>
-            `${k}@${packageLockJson?.dependencies[k]?.version ?? packageLockJson?.devDependencies[k]?.version ?? v}`
-        )
-      : ionicPackages.map(([k, v]) => `${k}@${v}`);
+    try {
+      packages = yarn ? await yarnPackages(sys, ionicPackages) : await npmPackages(sys, ionicPackages);
+    } catch (e) {
+      packages = ionicPackages.map(([k, v]) => `${k}@${v.replace('^', '')}`);
+    }
+
+    packages_no_versions = ionicPackages.map(([k]) => `${k}`);
 
-    return { packages };
+    return { packages, packages_no_versions };
   } catch (err) {
     hasDebug(config) && console.error(err);
-    return { packages };
+    return { packages, packages_no_versions };
   }
 }
 
+/**
+ * Visits the npm lock file to find the exact versions that are installed
+ * @param sys The system where the command is invoked
+ * @param ionicPackages a list of the found packages matching `@stencil`, `@capacitor`, or `@ionic` from the package.json file.
+ * @returns an array of strings of all the packages and their versions.
+ */
+async function npmPackages(sys: d.CompilerSystem, ionicPackages: [string, string][]) {
+  const appRootDir = sys.getCurrentDirectory();
+  const packageLockJson: any = await tryFn(readJson, sys, sys.resolvePath(appRootDir + '/package-lock.json'));
+
+  return ionicPackages.map(([k, v]) => {
+    let version = packageLockJson?.dependencies[k]?.version ?? packageLockJson?.devDependencies[k]?.version ?? v;
+    version = version.includes('file:') ? sanitizeDeclaredVersion(v) : version;
+    return `${k}@${version}`;
+  });
+}
+
+/**
+ * Visits the yarn lock file to find the exact versions that are installed
+ * @param sys The system where the command is invoked
+ * @param ionicPackages a list of the found packages matching `@stencil`, `@capacitor`, or `@ionic` from the package.json file.
+ * @returns an array of strings of all the packages and their versions.
+ */
+async function yarnPackages(sys: d.CompilerSystem, ionicPackages: [string, string][]) {
+  const appRootDir = sys.getCurrentDirectory();
+  const yarnLock = sys.readFileSync(sys.resolvePath(appRootDir + '/yarn.lock'));
+  const packageLockJson = sys.parseYarnLockFile(yarnLock);
+
+  return ionicPackages.map(([k, v]) => {
+    const identifiedVersion = `${k}@${v}`;
+    let version = packageLockJson.object[identifiedVersion]?.version;
```

Comment:
Since `object` is of type `any`, can we handle the case where it is `undefined`/`null`?

Reply:
We would handle that above, within the getInstalledPackages try/catch block.